### ISSUE TITLE
IIFE with leading semicolon wants to be intended when directly after imports

### ISSUE THREAD

**Issue by kjg** [eslint/eslint#11080]
**Tell us about your environment**

* **ESLint Version:** 4.18.2 
* **Node Version:** 8.9.3
* **npm Version:** 6.4.1

**What parser (default, Babel-ESLint, etc.) are you using?** Babel-ESLint

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
Standard.js default config
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
import $ from 'jquery'

// Bootstrap
;(() => {
  window.$ = $
})()

```

```bash
standard --parser babel-eslint
```

**What did you expect to happen?**
No error, no change of indention needed.

**What actually happened? Please include the actual, raw output from ESLint.**
It wants the `;(() => {` line of the iife with leading semicolon to be indented
```Expected indentation of 2 spaces but found 0.```

Interestingly if I move this IIFE to below other code, ESLint doesn't try to indent it, only when it's the first bit of code after the import statement.

**Are you willing to submit a pull request to fix this bug?**
I'm willing too, but I couldn't easily figure out exactly which part of the indent rule was causing this

**Comment by nzakas** [comment]
Thanks for the issue. Because you are using Standard, which is an opinionated code style guide, I'd suggest that you open an issue on that repo to discuss this issue.

Thanks!
